src/tokens.rs

```diff
@@ -229,7 +227,7 @@ impl<'a> Tokenizer<'a> {
     }
 
     fn comment_token(&mut self, start: usize) -> Token<'a> {
-        while let Some((_, ch)) = self.chars.clone().next() {
+        for (_, ch) in self.chars.clone() {
```

Comment:
This and the one below are semantically different, the current incarnation should remain.